src/netbase.cpp

```diff
@@ -347,15 +349,15 @@ enum class IntrRecvError {
  *      Sockets can be made non-blocking with SetSocketNonBlocking(const
  *      SOCKET&, bool).
  */
-static IntrRecvError InterruptibleRecv(uint8_t* data, size_t len, int timeout, const SOCKET& hSocket)
+static IntrRecvError InterruptibleRecv(uint8_t* data, size_t len, int timeout, const Sock& sock)
```

Comment:
Q: would it make sense to have this method as part of the `Sock`? (Something similar to `Sock::recv` but with a timeout parameter)

Reply:
Yes, indeed. That would be convenient together with a `send` method that retries upon temporary failures (see below).

I did not include such methods in this PR because it is self-sufficient as it is and achieves the initial purpose to introduce a mockable socket that can be used for fuzzing in https://github.com/bitcoin/bitcoin/pull/19203.

Possible future improvements (which I resisted adding in this PR):
* Add robust receiving method, like `InterruptibleRecv()` (or change that function to a method of the `Sock` class)
* Add robust sending method, like [`SendComplete()`](https://github.com/bitcoin/bitcoin/blob/2ae504c4b96013c87cc97909bdb00866fc4ab8a0/src/netbase.cpp#L1085) from https://github.com/bitcoin/bitcoin/pull/20685
* Use `Sock` in more places instead of the bare `SOCKET` - chase all places where `Sock::Release()` or `Sock::Get()` is called and see if the consumer can be changed to use `Sock` instead. E.g. in `CNode` constructor.